lint-hrx.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'fs';
+import p from 'path';
+import yargs from 'yargs/yargs';
+import fileLinesCount from 'file-lines-count';
+
+import {fromPath, SpecDirectory} from './lib-js/spec-directory';
+import RealDirectory from './lib-js/spec-directory/real-directory';
+import VirtualDirectory from './lib-js/spec-directory/virtual-directory';
+import {GithubActionsReporter} from './lib-js/reporter';
+
+/** The number of lines allowed in HRX files. */
+const LINE_THRESHOLD = 500;
+
+/**
+ * Verifies all HRX files in the project `spec` directory are under the
+ * {@link LINE_THRESHOLD}.
+ *
+ * HRX files are unwrapped and transformed into real files when
+ * {@link reporter.fix} is true.
+ *
+ * Skips HRX files if they include a top-level `options.yml` file annotated with
+ * :todo: or :ignore-for: `lint-hrx`. For example:
+ *
+ * ```hrx
+ * <===> options.yml
+ * # Ignored because <SOME REASON>
+ * :todo:
+ * - lint-hrx
```

Comment:
Might be a good idea to also allow sass-spec issue references, to match how we do TODOs for implementations.

Reply:
looking at how it's done, it just looks at whether it includes a substring, so `sass/libsass#123` doesn't do anything specific about being a github issue, it just happens to contain the `impl` value (libsass in this case)

With that in mind I updated it to contain `sass/sass-spec#`